src/org/traccar/database/DataManager.java

```diff
@@ -297,7 +297,7 @@ private void initDatabaseSchema() throws SQLException, LiquibaseException {
 
     public User login(String email, String password) throws SQLException {
         User user = QueryBuilder.create(dataSource, getQuery("database.loginUser"))
-                .setString("email", email)
+                .setString("email", email.trim().toLowerCase())
```

Comment:
Sorry, but neither of your points answer my question. Field is case-insensitive in the database, so why do we care what user enters and why do we need to lower-case it?